File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int r, l =0,res=0;
        for(r=0;r<prices.size();r++){
            if(prices[l]>prices[r]){
                l=r;
            }
            res = max(res,prices[r]-prices[l]);
        }
        return res;
    }
};

File: Data Structures & Algorithms/task-scheduling/submission-1.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] freq = new int[26];
        for(int i=0;i<tasks.length;i++){
            freq[tasks[i]-'A']++;
        }

        int max = 0;
        int count = 0;
        for(int i=0;i<26;i++){
            max = Math.max(freq[i],max);
        }

        for(int i=0;i<26;i++){
            if(freq[i]==max){
                count++;
            }
        }

        int calc = (max-1)*n + max + count-1;

        if(tasks.length>calc){
            return tasks.length;
        }else{
            return calc;
        }
    }
}
